torch/csrc/jit/codegen/cuda/ir_interface_nodes.h

```diff
@@ -326,6 +326,16 @@ class TORCH_CUDA_API TensorView : public Val {
     return memory_type_;
   }
 
+  void setMemoryType(MemoryType mt) {
+    memory_type_ = mt;
+    bool is_inp_or_out =
+        this->fusion()->hasInput(this) || this->fusion()->hasOutput(this);
+    if (is_inp_or_out)
```

Comment:
Could you add a check to make sure someone doesn't try to enable global memory for an intermediate tensor, as this isn't supported yet.